src/components/change-view.jsx

```diff
@@ -341,3 +370,30 @@ function changeMatches (change, other) {
 function isDisabled (element) {
   return element.disabled || element.classList.contains('disabled');
 }
+
+function relevantDiffTypes (versionA, versionB) {
+  let typeA = mediaTypeForVersion(versionA);
+
+  if (typeA.equals(mediaTypeForVersion(versionB))) {
+    return diffTypesFor(typeA);
+  }
+
+  // If we have differing types of content consider it an 'unkown' type.
+  return diffTypesFor(unknownType);
+}
+
+function mediaTypeForVersion (version) {
+  const contentType = version.content_type
+    || version.source_metadata.content_type;
+
+  if (contentType) {
+    return parseMediaType(contentType);
+  }
+
+  if (version.uri) {
+    const extension = version.uri.match(/^([^:]+:\/\/)?.*\/[^/]*(\.[^/]+)$/);
```

Comment:
This should probably fall back to the *page’s* URL if there was no extension in the version URI. It looks like we have quite a few .doc/.docx files where the raw version winds up with no extension, but the original URL had one.